packages/api/src/middleware/validators.ts

```diff
@@ -0,0 +1,62 @@
+import { RequestHandler } from "express";
+import validators from "../schema/validators";
+
+export const validatePost = (name: string): RequestHandler => {
+  const validate = validators[name];
+  if (!validate) {
+    throw new Error(`no validator found for ${name}`);
+  }
+
+  return (req, res, next) => {
+    const { body } = req;
+    if (!validate(body)) {
+      res.status(422);
+      return res.json({
+        errors: validate.errors.map((err) => JSON.stringify(err)),
+      });
+    }
+    next();
+  };
+};
+
+export const validateFormData = (name: string): RequestHandler => {
```

Comment:
Would it make sense to add a unit test for this? Or is it maybe already tested inside `generate.test.ts`?

Reply:
Yeah the easier way to test will be with a real API, with a schema. I can add some test cases on the `generate.test.ts` that checks these validations are working properly.